### DISCUSSION TITLE
Rename Files/Directories + Delete Directories

### DISCUSSION THREAD

**Discussion by tep** [coddingtonbear/obsidian-local-rest-api#123] [Ideas]
I've spun round-n-round trying to figure this out but, as far as I can tell, the API provides no way to rename a file or a directory -- and provides no way to delete a directory.

If this is already doable, please clue me in -- otherwise, this functionally should be added.

**Comment by coddingtonbear** [comment]
This module provides a [RESTful interface](https://en.wikipedia.org/wiki/REST), and unfortunately there are no REST semantics for renaming.  Instead, you have to delete (via `DELETE`) items at particular paths and re-create them at other paths (via `PUT`).   To "rename" a directory, just do the above for all items in whatever path.  Good luck!

**Reply by tep** [reply]
OK. I guess I'll fork your project instead.

**Reply by coddingtonbear** [reply]
godspeed!

**Comment by coddingtonbear** [comment]
Incidentally, you might find this helpful: https://github.com/coddingtonbear/obsidian-local-rest-api/wiki/Adding-your-own-API-Routes-via-an-Extension
